src/prefect/flows.py

```diff
@@ -2930,26 +2956,28 @@ def is_entrypoint_async(entrypoint: str) -> bool:
     Returns:
         True if the function is asynchronous, False otherwise.
     """
-    func_def, _ = _entrypoint_definition_and_source(entrypoint)
+    func_def, _, _ = _entrypoint_definition_and_source(entrypoint)
     return isinstance(func_def, ast.AsyncFunctionDef)
 
 
 def _entrypoint_definition_and_source(
     entrypoint: str,
-) -> Tuple[Union[ast.FunctionDef, ast.AsyncFunctionDef], str]:
+) -> Tuple[Union[ast.FunctionDef, ast.AsyncFunctionDef, ast.ClassDef], str, List[str]]:
```

Comment:
Could you add a docstring explaining what is returned from this function? It's not immediately clear what the last entry in the tuple is. It might also be a good idea to rename this function since it's returning more than just the entrypoint definition and source.